scripts/main/build.js

```diff
@@ -251,7 +261,10 @@ build.photo = function (data, disabled = false) {
 	html += lychee.html`
 			<div class='photo ${disabled ? `disabled` : ``}' data-album-id='${data.album_id}' data-id='${
 		data.id
-	}' data-tabindex='${tabindex.get_next_tab_index()}'>
+	}' data-tabindex='${tabindex.get_next_tab_index()}'
+			draggable='true'
+			ondragstart='lychee.startDrag(event)'
```

Comment:
Unfortunately, there are a lot of these kinds of errors, because our JS code is still not in a form which can be interpreted by the IDE in a reliable way. I am no expert on modern build chains for JS applications.

In the old times a JS application was mostly a bunch of files which were merged in the final step. This is the way how I am used to JS development and by co-incident this is also the way how the Gulpfile is written. Basically all JS files are merged together. Seemingly the build system dates to the same time when I turned my back to JS development.

As far as I understand this is not the way you would do it today. Instead one would define proper JS modules (keyword `module`) and then use these modules in other modules (via keyword `use`). In the build system one would not list the files individually, but the build process would collect all required files automatically.

This also seems the reason why the IDE is unhappy. For example, I get a lot of errors that `jQuery` aka `$` was not defined. (Of course it is, because we include the file `jquery.min.js` into the merge, but the IDE does not detect that.) I also get errors that an object of type `jQuery` cannot be assigned to an object of type `jQuery`. :smiley: The proper way would probably be to `use` jQuery.

I considered changing that and refactor out build system, but as I am a total noob in this respect and I could not find a quick solution, I dropped it again.

Long story short: For the moment we need to be selective about which IDE error we take seriously and which we can ignore. There are a lot of false positive. However, if anybody with more knowledge about JS and Gulp feels like fixing our build system and how we include modules into each other, I would be happy.

Reply:
> So we can ignore this for now and merge this PR?

I think another set of eyes should look over this PR before it's merged. I believe @nagmat84 said that he may be able to do it over the weekend?

Reply:
With respect to this point, yes. However, I made some other suggestions.